### PR TITLE
feat: Add option to use full window mode instead of using tech's fullscreen

### PR DESCRIPTION
**Pull request by mister-ben**
## Description
iPhone's native fullscreen isn't always desirable as you can't overlay controls or anything else. Adds an option to prefer "full window" mode over the video el's fullscreen.

## Specific Changes proposed
If the `preferFullWindow` option is set to true, on a browser that does not support the proper fullscreen API but does support fullscreen on the video element (i.e. iPhone), then `requestFullscreenHelper_()` should call `enterFullWindow()` instead of fullscreen on the tech.

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [ ] If necessary, more likely in a feature request than a bug fix
  - [x] Change has been verified in an actual browser (Chrome, Firefox, IE)
  - [x] Unit Tests updated or fixed
  - [x] Docs/guides updated
  - [ ] Example created ([starter template on JSBin](https://codepen.io/gkatsev/pen/GwZegv?editors=1000#0))
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

**Comment by gkatsev** [comment]
Would be it useful to also have an option that forces full window mode?

**Comment by mister-ben** [comment]
Maybe, I was only considering the case of overlays over "full screen" iOS really but we could cover that scenario too.

File: test/unit/player-fullscreen.test.js [hunk 1]
```diff
@@ -208,6 +208,26 @@ QUnit.test('fullscreenOptions from function args should override player options'
...
+
+  player.requestFullscreen();
+
+  assert.strictEqual(player.isFullscreen(), true, 'player considered fullscreen');
+  assert.strictEqual(player.isFullWindow, true, 'player is full window');
+
+  player.exitFullscreen();
+  assert.strictEqual(player.isFullWindow, false, 'full window is exited');
+  
+    player.dispose();
```
Comment (gkatsev) [inline test/unit/player-fullscreen.test.js:228]:
```suggestion

  player.dispose();
```

**Review by gkatsev — APPROVED** [review]

**Comment by hasanbuba** [comment]
8 Haz 2021 Sal, saat 18:55 tarihinde Gary Katsevman <
***@***.***> şunu yazdı:

**Pull request merged**